src/app/auth.guard.ts

```diff
@@ -33,9 +33,18 @@ export class AuthGuard implements CanActivate {
         if (result) {
           return true;  // Allow access
         }
+
+        
         // If no bypass in effect and user role is not in role restriction, then no access
-        if (roles.indexOf(JSON.parse(this.auth.getProfile()).app_metadata.roles) == -1)
```

Comment:
This line is checking for the index location of an array, `app_metadata.roles` in `roles`, which will always result to -1 since the roles array would only contain `ORGANIZATION`, `VOLUNTEER`, or `ADMIN`. As a result, the parent conditional statement would always return `false`.